src/vm/eventpipe.h

```diff
@@ -60,7 +60,11 @@ class EventPipe
     static EventPipeSession *GetSession(EventPipeSessionID id);
 
     // Specifies whether or not the event pipe is enabled.
-    static bool Enabled();
+    static bool Enabled()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return s_tracingInitialized && (s_numberOfSessions > 0);
```

Comment:
This might be a perf issue. This gets called as part of the check to see if logging is required at many sites in the runtime and that check is expected to be extremely fast when logging is off. The implicit cast of s_numberOfSessions to a uint32_t has a memory barrier inside of it which could be expensive. Given that you don't need that memory barrier for anything its probably easier to change this to s_numberOfSession.LoadWithoutBarrier() than to spend the time evaluating the performance to verify that it is/is not an issue in practice.

Reply:
`s_numberOfSessions` is protected for writes under the Crst, would it make more sense to use interlocked[increment|decrement] instead of making this member `Volatile`?

Reply:
In this case I think of the Volatile more as advertisement to other devs that the data is being used across multiple threads and a portion of that usage is lock-free. I'd recommend keeping it. Only looking at it from a correctness angle you could make this variable not a volatile and the behavior would still be accurate.

InterlockedIncrement/InterlockedDecrement ensure that the read,update,write are atomic, but the EventPipe lock is already providing an atomic update guarantee so there is no benefit gained using both.